From 0936907de4fb6286b463f14fe2ee1ca0414b901e Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Wed, 7 Aug 2024 11:47:26 +0200
Subject: [PATCH 0161/1010] net: dovetail: don't forward stolen skb heads to
 the core

Stolen skb heads are specific to the GRO process, oob buffers are
never part of it. So no need to pass those heads to recycle_oob_skb().

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 net/core/skbuff.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 11efdeed4f..29b1803e4f 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -1204,9 +1204,6 @@ void __kfree_skb_defer(struct sk_buff *skb)
 
 void napi_skb_free_stolen_head(struct sk_buff *skb)
 {
-	if (recycle_oob_skb(skb))
-		return;
-
 	if (unlikely(skb->slow_gro)) {
 		nf_reset_ct(skb);
 		skb_dst_drop(skb);
-- 
2.30.2

